src/logic/calculate.js

```diff
@@ -1,22 +1,113 @@
-import operate from './operate';
+import Operate from './operate';
 
-const calculate = ({ total, next, operation }, buttonName) => {
-  if (buttonName === '+/-') {
-    const negTotal = (total) * -1;
-    const negNext = (next) * -1;
-    return { total: negTotal, next: negNext };
-  }
-  if (buttonName === 'AC') {
-    return { total: 0, next: 0, operation: null };
-  }
-  if (buttonName === '.') {
-    return { total: '0.' };
-  }
-  if (buttonName === '%' || buttonName === '÷' || buttonName === 'X' || buttonName === '-' || buttonName === '+') {
-    const totalResult = operate(total, next, operation);
-    return { total: totalResult.toString() };
+const Calculate = ({ total, next, operation }, buttonName) => {
+  switch (buttonName) {
+    case 'AC':
+      return {
+        total: null,
+        next: null,
+        operation: null,
+      };
+    case '+/-':
+      if (next) {
+        return {
+          total,
+          next: Operate(next, '-1', 'x'),
+          operation,
+        };
+      }
+      return {
+        total: Operate(total, '-1', 'x'),
+        next,
+        operation,
+      };
+    case '0':
+    case '1':
+    case '2':
+    case '3':
+    case '4':
+    case '5':
+    case '6':
+    case '7':
+    case '8':
+    case '9':
```

Comment:
- [ ] you should have used regex to validate if it's a number instead of using 10 cases.